Source/player.cpp

```diff
@@ -1604,8 +1589,8 @@ void RemovePlrFromMap(int pnum)
 		}
 	}
 
-	for (y = 0; y < MAXDUNY; y++) {
-		for (x = 0; x < MAXDUNX; x++)
+	for (int y = 0; y < MAXDUNY; y++) {
+		for (int x = 0; x < MAXDUNX; x++) // NOLINT(modernize-loop-convert)

```

Comment:
It would try to apply a range look up to the loop which works out very poorly for 2D c-style arrays:
```cpp
for (y = 0; y < MAXDUNY; y++) 
    for (auto &x : dMonst)
        x[y] = monsterId;
```

So until we find a good replacement for c-style arrays in these situations the hint is unfortunately needed. The value of `modernize-loop-convert` in valid situations I think gratly out weighs this.
```cpp
for (auto &quest : quests) {
	quest._qactive = QUEST_NOTAVAIL;
}
```

Reply:
Got it. Yeah we'll need something like:

```cpp
for (auto tile in dungeon)
```

At some point.

The lack of abstraction kills this code so bad.